pkg/asset/agent/common.go

```diff
@@ -56,3 +58,24 @@ func IsSupportedPlatform(platform hiveext.PlatformType) bool {
 	}
 	return false
 }
+
+// DetermineReleaseImageArch returns the arch of the release image.
+func DetermineReleaseImageArch(pullSecret, pullSpec string) (string, error) {
+	templateFilter := "-o=go-template='{{if and .metadata.metadata (index . \"metadata\" \"metadata\" \"release.openshift.io/architecture\")}}{{index . \"metadata\" \"metadata\" \"release.openshift.io/architecture\"}}{{else}}{{.config.architecture}}{{end}}'"
+
+	var getReleaseArch = []string{
+		"oc",
+		"adm",
+		"release",
+		"info",
+		pullSpec,
+		templateFilter,
+	}
+	releaseArch, err := ExecuteOC(pullSecret, getReleaseArch)
```

Comment:
This is ok for this PR, but particularly with the heterogeneous cluster support coming in I'd like to see a follow-up switching to direct docker client calls as suggested in https://github.com/openshift/installer/pull/7349#discussion_r1349870697.
I think validation of the release payload arch is going to be critical for heterogeneous clusters in particular.

Reply:
@zaneb I agree, that something* needs to be done to validate the release image(s) for all IPI/assisted/agent installs. Not sure how that suggestion will work with sparse manifest list? But yeah, want to figure this out for heterogeneous in future releases.